packages/runner/src/run.ts

```diff
@@ -266,6 +266,51 @@ async function callCleanupHooks(runner: VitestRunner, cleanups: unknown[]) {
   }
 }
 
+/**
+ * Determines if a test should be retried based on its retryCondition configuration
+ */
+function shouldRetryTest(test: Test, errors: TestError[] | undefined): boolean {
+  const condition = test.retryCondition
+
+  // No condition means always retry
+  if (!condition) {
+    return true
+  }
+
+  // No errors means test passed, shouldn't get here but handle it
+  if (!errors || errors.length === 0) {
+    return false
+  }
```

Comment:
I think `errors` check should be done before any check that returns `true`.


```suggestion
  // No errors means test passed, shouldn't get here but handle it
  if (!errors || errors.length === 0) {
    return false
  }

  // No condition means always retry
  if (!condition) {
    return true
  }
```